pyomo/contrib/sensitivity_toolbox/tests/test_sens_unit.py

```diff
@@ -0,0 +1,806 @@
+                for i, (_, param, _, _) in
+                enumerate(sens.block._sens_data_list))
+        for param, ptb in zip(param_list, ptb_list):
+            for idx in param:
+                obj = param[idx]
+                var = param_var_map[obj]
+                con = param_con_map[obj]
+                self.assertEqual(instance.sens_state_value_1[var], ptb)
+                self.assertEqual(instance.DeltaP[con], obj.value-ptb)
+
+    @unittest.skipIf(not opt_ipopt.available(False), "ipopt is not available")
+    def test_get_dsdp1(self):
+        '''
+        It tests the function get_dsdp with a simple nonlinear programming example.
+        
+        min f: p1*x1+ p2*(x2^2) + p1*p2 
+         s.t c1: x1 = p1
+             c2: x2 = p2
+             c3: 10 <= p1 <= 10
+             c4: 5 <= p2 <= 5  
+        '''
+        variable_name = ['p1', 'p2']
+
+        m= ConcreteModel()
+        m.x1 = Var(initialize = 0)
+        m.x2 = Var(initialize = 0)
+        m.p1 = Var(initialize = 0)
+        m.p2 = Var(initialize = 0)
+        m.obj = Objective(expr = m.x1*m.p1+m.x2*m.x2*m.p2 + m.p1*m.p2 , sense=minimize)
+        m.c1 = Constraint(expr = m.x1 == m.p1)
+        m.c2 = Constraint(expr = m.x2 == m.p2)
+        theta= {'p1': 10.0, 'p2': 5.0}
+        for v in variable_name:
+            getattr(m, v).setlb(theta[v])
+            getattr(m, v).setub(theta[v])
+        dsdp, col = get_dsdp(m, variable_name, theta)
+        np.testing.assert_almost_equal(dsdp,[[-1.,  0., -1.,  0., -1.,  0.],
+                                                 [ 0., -1.,  0., -1.,  0., -1.]])
+
+        assert col == ['x1', 'x2', 'p1', 'p2']
+
+    @unittest.skipIf(not opt_ipopt.available(False), "ipopt is not available")
+    def test_get_dsdp2(self):
+        '''
+        It tests the function get_dsdp with rooney & biegler's model.
+        '''
+        variable_name = ['asymptote', 'rate_constant']
+
+        theta={'asymptote': 19.142575284617866, 'rate_constant': 0.53109137696521}
+        cov=np.array([[ 6.30579403, -0.4395341 ],[-0.4395341 ,  0.04193591]])
+        model_uncertain= ConcreteModel()
+        model_uncertain.asymptote = Var(initialize = 15)
+        model_uncertain.rate_constant = Var(initialize = 0.5)
+        model_uncertain.obj = Objective(expr = model_uncertain.asymptote*( 1 - exp(-model_uncertain.rate_constant*10  )  ), sense=minimize)
+        theta= {'asymptote': 19.142575284617866, 'rate_constant': 0.53109137696521}
+        for v in variable_name:
+            getattr(model_uncertain, v).setlb(theta[v])
+            getattr(model_uncertain, v).setub(theta[v])
+        dsdp, col =  get_dsdp(model_uncertain, variable_name, theta, {})
+        np.testing.assert_almost_equal(dsdp , [[-1.,  0., -1.,  0.],
+                                               [ 0., -1.,  0., -1.]])
+        assert col == ['asymptote', 'rate_constant']
+
+    
+    @unittest.skipIf(not opt_kaug.available(False), "k_aug is not available")
```

Comment:
As of somewhat recently, `SolverFactory("k_aug").available()` is returning false on my system, despite having `k_aug` on my path. @jsiirola @blnicho Any idea why this could be? Does k_aug now need to be registered somewhere?

Reply:
This appears to be a problem with `k_aug`.  We are currently using the default ASL interface for `k_aug` - and we recently made a change to the ASL interface to report "not available" if we couldn't parse the solver version string from `<solver> -v`.  Unfortunately, `k_aug` doesn't report anything that looks like a version number, and instead returns:
```
[K_AUG] written by D.T. @2018 (x86_64-pc-linux-gnu), ASL(20190605)

```

Reply:
Just opened a k_aug PR that will change the result of `k_aug -v` to be:
```console
K_AUG 0.1.0 (Linux x86_64), ASL(20180528)
```
I assume this will be able to be parsed?

Reply:
According to my local tests, which now run the k_aug-dependent tests, this should be fine.